pkg/clusteragent/admission/controllers/webhook/controller_base_test.go

```diff
@@ -62,3 +72,95 @@ func TestNewController(t *testing.T) {
 
 	assert.IsType(t, &ControllerV1beta1{}, controller)
 }
+
+func TestAutoInstrumentation(t *testing.T) {
+	tests := []struct {
+		name        string
+		pod         *corev1.Pod
+		config      string
+		expectPatch bool
+	}{
+		{
+			name:   "disabled webhooks should not patch",
+			config: "testdata/all_disabled.yaml",
+			pod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "foo",
+					Labels: map[string]string{
+						"admission.datadoghq.com/enabled": "true",
+					},
+					Namespace: "foo",
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name: "foo",
+						},
+					},
+				},
+			},
+			expectPatch: false,
+		},
+		{
+			name:   "disabled webhooks should not patch, even with targets",
+			config: "testdata/all_disabled_targets.yaml",
+			pod: &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "foo",
+					Labels: map[string]string{
+						"admission.datadoghq.com/enabled": "true",
```

Comment:
I think one of your testcases should be a billing service. given target/label matches.

Reply:
This one actually broke the test. Good call